### ISSUE TITLE
Support for nullable property in OpenAPI 3

### ISSUE THREAD

**Issue by Droidion** [elysiajs/elysia-swagger#86]
I have model defined like

```
t.Object({
  numericProp: t.Nullable(t.Number()),
})
```

This generates expected type `TUnion<[TNumber, TNull]>`.

And in OpenAPI I'm getting

```
"numericProp": {
  "anyOf": [
    {
      "type": "null"
    },
    {
      "type": "number"
    }
  ]
}
```

[OpenAPI spec](https://swagger.io/docs/specification/data-models/data-types/) says

```
OpenAPI 3.0 does not have an explicit null type as in JSON Schema, but you can use nullable: true to specify that the value may be null. Note that null is different from an empty string "".
```

So, the expected OpenAPI schema would be 

```
"numericProp": {
  "type": "null",
  "nullable": true
}
```

SwaggerUI also has problem rendering model example, saying `"numericProp": "Unknown Type: null",`

**Comment by bostjanpisler** [comment]
workaround I'm using for null

`t.Object({
  numericProp: t.Union([t.Number(), t.Null()])
})`

**Comment by Mudbill** [comment]
Came to report this same issue. You'd think that `t.Nullable(t.String())` would generate the correct type for the UI. Instead it appears as this:

<img width="327" alt="image" src="https://github.com/elysiajs/elysia-swagger/assets/8832665/e9426f79-9c5c-4ea6-a9cd-eb9bf6642091">

<img width="441" alt="image" src="https://github.com/elysiajs/elysia-swagger/assets/8832665/1f846be0-add6-4ede-915f-da18c0ee0077">

---

The workaround that I used was this:

```ts
nullableString: {
  ...t.String(),
  nullable: true
}
```

Problem here is that `nullable` is not recognized as a property, and therefore isn't type-checked. But at least it appears properly in the UI.

<img width="252" alt="image" src="https://github.com/elysiajs/elysia-swagger/assets/8832665/75741c40-2c28-4e4d-a435-71ff2b9e6d0c">

<img width="322" alt="image" src="https://github.com/elysiajs/elysia-swagger/assets/8832665/3d92e5a4-3cee-4e3c-b323-c61bffe45fee">

I think `t.Nullable` should be changed to use the `nullable` flag instead of merging with a union of null, at least if the appropriate version of OpenAPI is specified. The merging union seems to be appropriate for a different version of the spec than the default UI version expects.

**Comment by Mudbill** [comment]
Okay so I discovered something in plain sight that actually seems to have a big impact.

Since the null union is a OpenAPI 3.1 feature (not 3.0), it is of course important that SwaggerUI is parsing 3.1. However, the default seems to be 3.0.

<details><summary>Default and incorrect setup</summary>
<p>

Notice how it says "OAS 3.0" in the green badge up top of the web page.

```ts
const app = new Elysia()
  .use(swagger())
  .post("/", () => "Hello Elysia", {
    body: t.Object({
      nullableString: t.Nullable(t.String()),
    }),
  })
  .listen(3000);
```

<img width="572" alt="image" src="https://github.com/elysiajs/elysia-swagger/assets/8832665/f048f308-8abc-4536-b1c6-07640cf72754">

</p>
</details> 


Changing the OpenAPI version for Swagger seems to make it handle the correct types that `elysia.t` generates.

<details><summary>Corrected version matching</summary>
<p>

```ts
const app = new Elysia()
  .use(
    swagger({
      documentation: {
        openapi: "3.1.0", // Add this
      },
    })
  )
  .post("/", () => "Hello Elysia", {
    body: t.Object({
      nullableString: t.Nullable(t.String()),
    }),
  })
  .listen(3000);
```

<img width="603" alt="image" src="https://github.com/elysiajs/elysia-swagger/assets/8832665/692c4181-b545-4ddb-acb6-044ce8b1200f">

<img width="408" alt="image" src="https://github.com/elysiajs/elysia-swagger/assets/8832665/5f8dbab1-4b76-49f4-b314-6d0ac6572731">

</p>
</details>

**Comment by Droidion** [comment]
Yes, @Mudbill has the proper good workaround, worked for me. I also agree, that maybe `3.1.0` should be default version, and/or docs can say something about recommended version.
